BITSWAP.md

```diff
@@ -1,200 +1,200 @@
 
-Bitswap is a message based protocol, as opposed to response-reply. All messages contain wantlists, or blocks. Upon receiving a wantlist, an IPFS node should consider sending out wanted blocks if it has them. Upon receiving blocks, the node should send out a notification called a `Cancel` signifying that they no longer want the block. At the protocol level, Bitswap is very simple.
+Bitswap is a relatively simple protocol, where complexity lives in the particular implementations to optimize for their particular balance of aspects such as throughput, latency, fairness, memory usage, etc.
 
-While Bitswap is a relatively simple protocol, a time- and memory- performant implementation requires that many details be carefully thought out. We aim to document these details here so that future implementers may build upon the knowledge gathered over the several iterations of Bitswap.
+# Bitswap Protocol Versions
 
-# Subsystems
+There are multiple Bitswap versions and more may evolve over time. We give brief overviews as to the changes behind each protocol version.
 
-There are two primary flows that Bitswap manages: requesting blocks from and serving blocks to peers. Block requests are primarily mediated by the want-manager, which tells our peers whenever we want a new block. Serving blocks is primarily handled by the decision engine, which decides how resources should be allocated among our peers.
+- `/ipfs/bitswap/1.0.0` - Initial version
+- `/ipfs/bitswap/1.1.0` - Support CIDv1
+- `/ipfs/bitswap/1.2.0` - Support Wantlist Have's and Have/DontHave responses
 
-The subsystems involved in these flows are detailed in the following subsections.
+## Bitswap 1.0.0
 
-**TODO**: Update graphic
+### Interaction Pattern
 
-![](https://cloud.githubusercontent.com/assets/1211152/21071077/4620387a-be4a-11e6-895c-aa8f2b06aa4e.png)
+Given that a client C wants to fetch data from some server S:
 
-## Types
-
-The following types are used in the descriptions of the Bitswap subsystems.
-
-  - `CID`: A [content-addressed identifier](https://github.com/ipld/cid) that refers to a particular `Block`.
-  - `Peer`: Another Bitswap instance that we are connected to.
-  - `Block`: A binary blob.
-  - `Message`: A Bitswap message.
-  - `Entry`: A wantlist entry that may be included in a Message when adding/removing a particular `CID` from our wantlist. Contains:
-      - `CID` referring to a particular block.
-      - `Priority` relative priority with which the user wants `CID` (relevant only if `Cancel` is not true.
-      - `Cancel` is a boolean representing whether this `Entry` is meant to remove `CID` from our wantlist.
-  - `Ledger`: A record of the aggregate data exchanged between two peers. Each peer stores one `Ledger` for each of their peers.
+1. C opens a stream `s_want` to S and sends a message for the blocks it wants
+    1. C may either send a complete wantlist, or an update to an outstanding wantlist
+    2. C may reuse this stream to send new wants
```

Comment:
Maybe, I just want to make sure that people don't think they need to (or that they cannot) reuse the stream to send wants rather than opening a stream, sending a message, and the closing the stream each time.